### ISSUE TITLE
simdMat4 is not part of the namespace

### ISSUE THREAD

**Issue by giordi91** [g-truc/glm#505]
Hi, I was trying to enable avx instruction for matrix mult for skinning operation, but looks like at compile time is no t able to find simdMat4, saying that is not part of the namespace. I am using the latest trunk on VS 2015 win 10. 

Those ore the includes:
#include <glm/glm.hpp>
#include <glm/gtc/matrix_transform.hpp>
#include <glm/gtc/quaternion.hpp>
#include <glm/gtc/random.hpp>
#include <glm/gtx/simd_vec4.hpp>
#include <glm/gtx/simd_mat4.hpp>

And I am just trying to instantiate it in a struct:
struct Joint
	{
		glm::simdMat4 m_inv_bind_pose;
		string m_name;
		unsigned char m_parent_id;
	};

I found this example on the net:
https://classes.soe.ucsc.edu/cmps160/Spring13/projects/whe3/proj/glm/test/gtx/gtx_simd_mat4.cpp
And looks like I am doing the same thing but does not compile my code. Any Idea on what I might be doing wrong?

M.

**Comment by bcrist** [comment]
Are you compiling an x64 project with GLM 0.9.7.4 (or earlier)?

GLM detects SSE2 on MSVC x86 by seeing if `_M_IX86_FP = 2`.  SSE2 is a standard part of x86-64, but MSVC doesn't define that symbol for x64 builds, so GLM will fall back to `GLM_ARCH_PURE`, which makes pretty much everything in `gtx_simd_*` go away.

This issue looks to have been fixed in master, but hasn't made it into a release as of this writing.

As a workaround you can use `GLM_FORCE_SSE2` (or `GLM_FORCE_SSE3`, etc).

**Comment by giordi91** [comment]
@bcrist  Hi thanks for getting me back to me, yes I am compiling on VS 2015 x64 and with the latest one i could find, 0.9.7.4.  I was actually setting force GLM_FORCE_AVX, since  I have those instruction available to my machine. but I don't think it worked. BUT I most likely messed up on my side, let me try again and i will get back to you. Thanks a lot

M.

**Comment by giordi91** [comment]
@bcrist  I gave it another go, still no luck, I tried with the force , and setting also visual studio to use avx, sse2 etc no luck , here is a snippet of the code:

![simd](https://cloud.githubusercontent.com/assets/2525797/15242832/21bc7de0-18f0-11e6-9682-3ecc6a6d2117.jpg)

am I doing anything wrong? To note the screenshot says SSE but I tried with SSE2 and avx aswell no luck.
I found that the definition stuff actually affects the setup.hpp, do i need to include something specific in order to have that header correctly show up in the code? I tried to include directly setup.hpp before anything (but after the define, and no luck). 
Any idea?

**Comment by bcrist** [comment]
`GLM_FORCE_SSE` doesn't appear to be a symbol that is checked (at least in 0.9.7.4), but `GLM_FORCE_SSE2` or `GLM_FORCE_AVX` should both do the trick.  I'm using `GLM_FORCE_SSE4` and have no problems finding `simdMat4` in the `glm` namespace.

The only other difference I see is that you're defining those switches in code instead of specifying them directly to the compiler through project configuration -> C/C++ -> Preprocessor -> Preprocessor Definitions.  But if you were including GLM in other translation units without the same `#define`s msvc would likely give you an earful about redefinition of symbols.

**Comment by giordi91** [comment]
yes thank you , doing the define at project level did the trick! thanks a lot.

M
